### ISSUE TITLE
Legal question

### ISSUE THREAD

**Issue by amirpdlll** [Nerixyz/instagram_mqtt#24]
Hello ,
Using libraries such as this library that are making reverse engineering on instagram is legal ? can we make applications for our customers using these libraries or not ?

thanks

**Comment by Androz2091** [comment]
No, that's not allowed by Instagram. A similar repository was taken down via a DMCA request by Facebook.

Some links for more informations:
https://news.ycombinator.com/item?id=22209892
https://github.com/dilame/instagram-private-api/issues/1066
https://github.com/github/dmca/blob/master/2020/01/2020-01-22-facebook.md

**Comment by amirpdlll** [comment]
Oh thank you , so why the have blocked mgp25 instagram library but not libraries such as dilame/instagram-prive-library ?

**Comment by amirpdlll** [comment]
> No, that's not allowed by Instagram. A similar repository was taken down via a DMCA request by Facebook.
> 
> Some links for more informations:
> https://news.ycombinator.com/item?id=22209892
> [dilame/instagram-private-api#1066](https://github.com/dilame/instagram-private-api/issues/1066)
> https://github.com/github/dmca/blob/master/2020/01/2020-01-22-facebook.md

Oh thank you , so why the have blocked mgp25 instagram library but not libraries such as dilame/instagram-prive-library ?

**Comment by Androz2091** [comment]
🤷‍♂ They can't block everything, and the mgp25's library was more known

**Comment by amirpdlll** [comment]
> 🤷‍♂ They can't block everything, and the mgp25's library was more known. I think that's why

Thank you , so I won't use libraries such as that to respect instagram rules

thank you

**Comment by sergeykomlev** [comment]
> > 🤷‍♂ They can't block everything, and the mgp25's library was more known. I think that's why
> 
> Thank you , so I won't use libraries such as that to respect instagram rules
> 
> thank you

You can try official [Instagram Basic Display API](https://developers.facebook.com/docs/instagram-basic-display-api/) . 

That’s not very hard to get access to API, if you don’t do any illegal.

Complete request for permissions, attach screenshots and test login data. If all will be ok, Facebook will approve your app.

**Comment by Nerixyz** [comment]
@sergeykomlev The basic display API is as the name suggests **basic**, so if you want to do anything beyond getting post and/or user info, you either need to register a new app at Facebook (and probably be a business partner) or you reverse engineer your own library. 

And that's especially true for anything realtime related (which is what this repository is about).
